tests/test_cliPluginLoader.py

```diff
@@ -59,19 +65,96 @@ def cli():
+                                         "create": ["lsst.daf.butler.cli.cmd"]})
+
+    def test_getPluginCommands(self):
+        """Test getting the plugin commands.
+
+        Normally daf_butler plugins are loaded without using the environment
+        variable, but it's convenient to use them here to verify the plugin
+        path works."""
+        with patch.dict("os.environ", {"DAF_BUTLER_PLUGINS": "lsst.daf.butler.cli.cmd"}):
+            pluginCommands = butler.LoaderCLI._getPluginCommands()
+            self.assertEqual(pluginCommands, {"config-dump": ["lsst.daf.butler.cli.cmd"],
+                                              "config-validate": ["lsst.daf.butler.cli.cmd"],
+                                              "create": ["lsst.daf.butler.cli.cmd"]})
+
+    def test_mergeCommandLists(self):
+        """Verify dicts of command to list-of-source-package get merged
+        properly."""
+        self.assertEqual(butler.LoaderCLI._mergeCommandLists({"a": [1]}, {"b": [2]}), {"a": [1], "b": [2]})
+        self.assertEqual(butler.LoaderCLI._mergeCommandLists({"a": [1]}, {"a": [2]}), {"a": [1, 2]})
+
+    def test_listCommands_duplicate(self):
+        """Test getting the plugin commands in a situation where duplicate
+        commands are present.
+
+        Normally daf_butler plugins are loaded without using the environment
+        variable. In this case we get the commands by both paths, which causes
+        daf_butler commands to be sourced from two locations, causing a
+        failure."""
+        loaderCli = butler.LoaderCLI()
+        with patch.dict("os.environ", {"DAF_BUTLER_PLUGINS": "lsst.daf.butler.cli.cmd"}):
+            with self.assertRaises(click.ClickException, msg="Command 'create' exists in packages "
+                                   "lsst.daf.butler.cli.cmd, lsst.daf.butler.cli.cmd. "
+                                   "Command 'config-dump' exists in packages "
+                                   "lsst.daf.butler.cli.cmd, lsst.daf.butler.cli.cmd. "
+                                   "Command 'config-validate' exists in packages "
+                                   "lsst.daf.butler.cli.cmd, lsst.daf.butler.cli.cmd. "
+                                   "Duplicate commands are not supported, aborting."):
+                loaderCli.list_commands(None)
+
+    def test_getCommand_local(self):
+        """Test the get_command function for commands in daf_butler."""
+        self.assertEqual(butler.LoaderCLI().get_command(None, "create"), cmd.create)
+
+    def test_getCommand_plugin(self):
+        """Test the get_command function for commands loaded as plugins.
+
+        Typically plugin commands are not in daf_butler, but here we mask the
+        local commands in the TestCLI class and expose them via the environment
+        variable."""
+        with patch.dict("os.environ", {"DAF_BUTLER_PLUGINS": "lsst.daf.butler.cli.cmd"}):
```

Comment:
Did not know about this method. I wrote `modified_environment` context manager in test_config.py that does exactly that.  I guess I need to change test_config to use unittest.mock.